implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalFxMeterRegistry.java

```diff
@@ -170,23 +192,26 @@ protected void publish() {
         );
     }
 
-    private Stream<SignalFxProtocolBuffers.DataPoint.Builder> addTimeGauge(TimeGauge timeGauge) {
+    Stream<SignalFxProtocolBuffers.DataPoint.Builder> addTimeGauge(TimeGauge timeGauge) {
         return Stream.of(addDatapoint(timeGauge, GAUGE, null, timeGauge.value(getBaseTimeUnit())));
     }
 
-    private Stream<SignalFxProtocolBuffers.DataPoint.Builder> addGauge(Gauge gauge) {
+    Stream<SignalFxProtocolBuffers.DataPoint.Builder> addGauge(Gauge gauge) {
+        if (fixHistogramBucketsType && gauge.getId().getName().endsWith(".histogram")) {
```

Comment:
Nit: the gauges created by `HistogramGauges` are always synthetic, adding that to the condition can make it a bit more foolproof:
```suggestion
        if (fixHistogramBucketsType
                && gauge.getId().syntheticAssociation() != null
                && gauge.getId().getName().endsWith(".histogram")) {
```